hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,329 @@
+
+        # Reconstructing reference wave and applying it (division):
+        if reference is None:
+            wave_reference = 1
+        elif reference.axes_manager.navigation_size != self.axes_manager.navigation_size:  # case when reference is 1d
+            wave_reference = reference.deepcopy()
+
+            # Prepare parameters for reconstruction of the reference wave:
+
+            if reference.axes_manager.navigation_size != sb_position.axes_manager.navigation_size:  # 1d reference, but
+                # parameters are multidimensional
+                sb_position_ref = sb_position.inav[0].data
+            else:
+                sb_position_ref = sb_position_temp
+
+            if reference.axes_manager.navigation_size != sb_size.axes_manager.navigation_size:  # 1d reference, but
+                # parameters are multidimensional
+                sb_size_ref = np.float64(sb_size.inav[0].data)
+            else:
+                sb_size_ref = sb_size_temp
+
+            if reference.axes_manager.navigation_size != sb_smoothness.axes_manager.navigation_size:  # 1d reference, but
+                # parameters are multidimensional
+                sb_smoothness_ref = np.float64(sb_smoothness.inav[0].data)
+            else:
+                sb_smoothness_ref = sb_smoothness_temp
+            #
+
+            wave_reference.map(reconstruct, holo_sampling=self.sampling, sb_size=sb_size_ref,
+                               sb_position=sb_position_ref, sb_smoothness=sb_smoothness_ref, output_shape=output_shape,
+                               plotting=plotting, show_progressbar=show_progressbar)
+
+        else:
+            wave_reference = reference.deepcopy()
+            wave_reference.map(reconstruct, holo_sampling=self.sampling, sb_size=sb_size_temp,
+                               sb_position=sb_position_temp, sb_smoothness=sb_smoothness_temp,
+                               output_shape=output_shape, plotting=plotting, show_progressbar=show_progressbar)
+
+        wave_image = wave_object / wave_reference
+
+        wave_image.set_signal_type('electron_wave')  # New signal is a wave image!
+
+        wave_image.axes_manager.signal_axes[0].scale = self.sampling[0] * self.axes_manager.signal_shape[0] / \
+                                                       output_shape[0]
+        wave_image.axes_manager.signal_axes[1].scale = self.sampling[1] * self.axes_manager.signal_shape[1] / \
+                                                       output_shape[1]
+
+        # Reconstruction parameters are stored in holo_reconstruction_parameters:
+        rec_param_dict = OrderedDict([('sb_position', sb_position_temp), ('sb_size', sb_size_temp),
+                                      ('sb_units', sb_unit), ('sb_smoothness',
+                                                              sb_smoothness_temp)])
+
+        wave_image.metadata.Signal.add_node('holo_reconstruction_parameters')
```

Comment:
I am not sure if it is a good idea to store the reconstruction parameters as they may be large. These days we tend to store only what is strictly needed for posterior analysis. I suppose that you want to store them for log purposes. We don't do this for any of the other methods to avoid adding clutter to the signal. Could at least this be optional with default `False`? Beside, what about storing it in `Signal.Holography.Reconstruction_parameters`? Notice that we capitalize node names.